storage/local/series.go

```diff
@@ -339,52 +339,47 @@ func (s *memorySeries) preloadChunks(
 				cd.unpin(mss.evictRequests)
 			}
 			chunkOps.WithLabelValues(unpin).Add(float64(len(pinnedChunkDescs)))
-			return nil, err
+			return nil, nopIter, err
```

Comment:
Right now, we could also just return nil. This is essentially in preparation of later changes (being worked on in branch beorn7/storage4). Errors bubbling up here are triggered by data corruption. In the future change, corrupted series will be quarantined, and no error will bubble up out of storage/local. Instead, a nopIter will be returned in the same way as for a series that got deleted or doesn't exist normally.